### DISCUSSION TITLE
Passkey - Disable authentication with another device

### DISCUSSION THREAD

**Discussion by BartekMilew** [passkeydeveloper/community#5] [Passkeys on the Web Q&A]
I would ask about opportunity to disable and avoid opportunity for user to authenticate with another device.
I encounter information "Sign in with another device" during authentication process and I want to prevent it from appearing. Is it possible ? We want to disable this option for our users. I see that there is option during registration only with attachment-platform

**Comment by timcappalli** [comment]
Could you share more about the use case? If the user has a passkey on another authenticator, they should be allowed to use it.

**Reply by BartekMilew** [reply]
For registration I use authenticatorAttachment „platform” to force user to use authenticator from his device.  So I know he has passkey on his device. For authentication there isn’t option to use authenticatorAttachment „platform”. We are adding internal transport but now enough. We don’t want to have „sign in with another device”. Is it possible ?

**Reply by timcappalli** [reply]
Right but what's the use case for restricting which authenticators a user can use a passkey from?

If the user has the passkey on their phone or a security key, why prevent them from using it? It's a bit user hostile.

**Reply by BartekMilew** [reply]
Requirement of the product I am working on.  I am trying to find solution for this

**Reply by timcappalli** [reply]
I'm not aware of any WebAuthn client that would explicitly allow this behavior.

**Reply by dasali-jenario** [reply]
Isn't there an option to limit the authenticators using the AutofillUI / Conditional UI? 

If a "platform-attached" passkey is available, there's no point in offering the user options like cross-device login, so filtering that option out could make sense. This doesn't mean using cross-device for authentication is bad, but if I want to reduce the options that can lead to user confusion, such a limitation could make sense.

**Reply by timcappalli** [reply]
> Isn't there an option to limit the authenticators using the AutofillUI / Conditional UI?

No, there's not

> If a "platform-attached" passkey is available, there's no point in offering the user options like cross-device login, so filtering that option out could make sense. This doesn't mean using cross-device for authentication is bad, but if I want to reduce the options that can lead to user confusion, such a limitation could make sense.

If a passkey is available in a local authenticator, most client UIs will prefer that, but still offer an "escape hatch" to CDA, as the user may have more than one account, and the passkey for that account may be on another device.

**Reply by dasali-jenario** [reply]
If an RP wants to have a more consistent flow across clients, they would then not use AutoFillUI and have the user type their identifier, hoping they provide a valid one. 

I get a lot of pushback from stakeholders in a project I support on the fact that using passkey on different clients looks so different. It's a difficult issue to navigate for people who are used to control the user-journeys end-to-end.

Appologies if I derail this conversation. I think I understand where the desire of the initial question comes from. The answer seems to be "no"?

**Reply by timcappalli** [reply]
Autofill UI has been tested heavily with users and has the highest success rate for sign ins. I highly recommend you use it.

> that using passkey on different clients looks so different. It's a difficult issue to navigate for people who are used to control the user-journeys end-to-end.

Did you get the same push back for the password autofill experiences that vary slightly on each client? Then you have OTP and magic link experiences which are significantly different across the board with tons of variables.

**Reply by dasali-jenario** [reply]
> Did you get the same push back for the password autofill experiences that vary slightly on each client? Then you have OTP and magic link experiences which are significantly different across the board with tons of variables.

No because the password manager is not deployed or mandated by them. They get a lot of customers calling their support-line because they struggle with password managers, but since they're not part of the service they don't have to support them.

**Reply by timcappalli** [reply]
So then I assume you've disabled autofill on your login forms today?

**Reply by dasali-jenario** [reply]
This is a net-new product my customer is working on and their current identity solutions are based on username & password, and they probably didn't disable anything. 

I see your point. There's just a different expectation with passkey than with passwords. Happy to have a chat offline or in a different forum.

**Reply by madelmiro2-code** [reply]
> Could you share more about the use case? If the user has a passkey on another authenticator, they should be allowed to use it.

Yes he does one sec, im terrible here i dont know anything

**Reply by madelmiro2-code** [reply]
> Could you share more about the use case? If the user has a passkey on another authenticator, they should be allowed to use it.

![Screenshot_20250923_220031_Samsung Internet](https://github.com/user-attachments/assets/3331ef72-c39a-4037-8c44-b5105f62b2e2)
![Screenshot_20250923_220000_Samsung Internet](https://github.com/user-attachments/assets/7e893771-1876-4c62-bd6b-2cd3a81e5780)
That maybe??

**Comment by ajmalbaburi071-wq** [comment]
![IMG-20260321-WA0012](https://github.com/user-attachments/assets/ee9a0e6a-bdfd-4510-9133-bbe6fc815c00)
